### ISSUE TITLE
Issue with setExternalUserID

### ISSUE THREAD

**Issue by Droppix** [OneSignal/OneSignal-iOS-SDK#865]
**Description:**
This is my code:

```
        OneSignal.setRequiresUserPrivacyConsent(true)
        OneSignal.consentGranted(true)

       let onesignalInitSettings : [String:Any] = [
            kOSSettingsKeyAutoPrompt: false,
            kOSSettingsKeyInAppLaunchURL: true,
            kOSSettingsKeyInFocusDisplayOption : OSNotificationDisplayType.notification.rawValue
        ]
        OneSignal.initWithLaunchOptions(launchOptions,
                                        appId: ONESIGNEL_APPID,
                                        handleNotificationReceived: notificationReceived,
                                        handleNotificationAction: notificationAction,
                                        settings: onesignalInitSettings)
        OneSignal.inFocusDisplayType = OSNotificationDisplayType.notification
        OneSignal.add(self as OSPermissionObserver)
        
        OneSignal.setExternalUserId("123) // Same with callback
        OneSignal.promptForPushNotifications(userResponse: { accepted in
            if accepted {
                OneSignal.setSubscription(subscribe)
                if subscribe == true {
                    //OneSignal.setExternalUserId(customerID)
                }
                else {
                    OneSignal.removeExternalUserId()
                }
                
                self.updateServerPlayerID()
            }
        })
```

1/ First remark
**EXTERNAL USER ID** never appear in the dashboard  (I choose my columns: External User ID, ...)
The field is always BLANK

2/ ERROR :  ["All included players are not subscribed"]
Send push with Onesignal client (with nodejs) https://github.com/zeyneloz/onesignal-node/tree/v2.x
I try send notifcation via target 'include_external_user_ids' and include_players_ids (include_players_ids works!)

```
const oneSignalClient = new oneSignal.Client({      
  userAuthKey: userAuthKey,
  app: { appID, apiKey }
})

json =  {
          "headings":{"en":"xxxx"},
          "contents":{"en":"yyyy"},
          "include_external_user_ids":["123"],"data":{"ticket_id":"5"},"priority":10}}

 let notif = new oneSignal.Notification(json)
 oneSignalClient.sendNotification(notif, function (error, httpResponse, data) {
console.log(data.errors) // I HAVE ERROR :  ["All included players are not subscribed"]

 }
```

So, I try to emit notification with 
**Environment**
OneSignal.framework : 2.15.4 (use carthage) or SPM 2.16.x

I've spent a lot of time searching, reading the documentation, but no clue, so if you have an idea?
Thanks

**Comment by shalinipk** [comment]
Seeing the issue on 3.2.1 as well

**Comment by VasylRomanets-MoonActive** [comment]
@Droppix, @shalinipk,  hi! I also had the problem on iOS but with OneSignal Unity SDK. After calling `OneSignal.setExternalUserId()` API "EXTERNAL USER ID" dashboard field was empty. In my case I solved the problem by calling the `OneSignal.setExternalUserId()` after user accepted push notifications with the default system prompt.

**Comment by shalinipk** [comment]
Update - I filed a support ticket with OneSignal. They were extremely responsive - there was a bug in their code. Fix is in OneSignal release  3.3.0

Here is my code - seems to work now.

```
OneSignal.setEmail(userEmail) {
OneSignal.setExternalUserId(currentUser.getId())
} withFailure: { (error) in
if let error = error {
log("OneSignalManager: \(error.localizedDescription)")
}
}
```

Hope this helps

**Comment by rgomezp** [comment]
This is no longer an issue. Please upgrade to 3.3.0!

Enjoy!
